cylc/flow/parsec/fileparse.py

```diff
@@ -243,23 +263,42 @@ def read_and_proc(fpath, template_vars=None, viewcfg=None, asedit=False):
     if do_empy:
         if flines and re.match(r'^#![Ee]m[Pp]y\s*', flines[0]):
             LOG.debug('Processing with EmPy')
+            tvars = copy(template_vars)
+            if rose_vars['empy:suite.rc'] is not None:
+                for key, value in rose_vars['empy:suite.rc'].items():
+                    tvars[key] = literal_eval(value)
             try:
                 from cylc.flow.parsec.empysupport import empyprocess
             except (ImportError, ModuleNotFoundError):
                 raise ParsecError('EmPy Python package must be installed '
                                   'to process file: ' + fpath)
-            flines = empyprocess(flines, fdir, template_vars)
+            flines = empyprocess(flines, fdir, tvars)
 
     # process with Jinja2
     if do_jinja2:
         if flines and re.match(r'^#![jJ]inja2\s*', flines[0]):
             LOG.debug('Processing with Jinja2')
+            tvars = copy(template_vars)
+            if rose_vars['jinja2:suite.rc'] is not None:
+                for key, value in rose_vars['jinja2:suite.rc'].items():
+                    tvars[key] = literal_eval(value)
             try:
                 from cylc.flow.parsec.jinja2support import jinja2process
             except (ImportError, ModuleNotFoundError):
                 raise ParsecError('Jinja2 Python package must be installed '
                                   'to process file: ' + fpath)
-            flines = jinja2process(flines, fdir, template_vars)
+            flines = jinja2process(flines, fdir, tvars)
+
+    # Inject rose-suite.conf environment variables into the workflow env.
+    # Appended to ensure that Rose settings over-ride flow.cylc.
+    if rose_vars['env'] is not None:
+        env_lines = [
+            f'        {key} = {value}' for
+            key, value in rose_vars['env'].items()
+        ]
+        flines += ['[scheduler]']
```

Comment:
This seems a tad crude - anyone suggest a better way?

Reply:
append/extend the list?
```python
        flines.append('[scheduler]')
        flines.append('    [[environment]]')
        flines.extend( '        %s = %s' % keyval for keyval in rose_vars['env'].items())
```

Reply:
I meant less crude modifying the end of the flow.cylc file. But yes.

Reply:
I figured that given the INI structure that appending was safer...